### ISSUE TITLE
[Bug] mac build process failed

### ISSUE THREAD

**Issue by kousaku-maron** [AppFlowy-IO/AppFlowy#80]
**Describe the bug**
I click app_flowy button on vscode.

### occuring error

```
lib/main.dart:1
--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:macOS, arch:x86_64, id:<-- my-pc-id --> }
{ platform:macOS, name:Any Mac }
Undefined symbols for architecture x86_64:
  "_link_me_please", referenced from:
      static flowy_sdk.FlowySdkPlugin.dummyMethodToEnforceBundling() -> () in flowy_sdk(FlowySdkPlugin.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
note: Using new build system
note: Planning
note: Build preparation complete
note: Building targets in parallel
** BUILD FAILED **

Exception: Build process failed
```

flutter doctor -v

```
[✓] Flutter (Channel dev, 2.6.0-11.0.pre, on macOS 11.6 20G165 darwin-x64,
    locale ja-JP)
    • Flutter version 2.6.0-11.0.pre at /Users/<-- my-pc-username -->/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4b330ddbed (10 weeks ago), 2021-09-16 17:29:58 -0700
    • Engine revision 5b81c6d615
    • Dart version 2.15.0 (build 2.15.0-116.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/<-- my-pc-username -->/Library/Android/sdk
    • Platform android-30, build-tools 29.0.2
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 49.0.1
    • Dart plugin version 192.8052
    • Java version OpenJDK Runtime Environment (build
      1.8.0_212-release-1586-b4-5784211)

[✓] IntelliJ IDEA Community Edition (version 2019.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.62.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.28.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 11.6 20G165 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 95.0.4638.69

• No issues found!
```

**Comment by Malenconiaprincep** [comment]
+ 1

**Comment by RanKKI** [comment]
See here https://github.com/AppFlowy-IO/appflowy/issues/68

**Comment by appflowy** [comment]
@kousaku-maron Still have any problem?

**Comment by xueqiliu** [comment]
@appflowy  I also encountered the same problem and did not solve it

**Comment by paranoea** [comment]
@appflowy
I also encountered the same problem and did not solve it
rustc version：rustc 1.59.0-nightly (48a5999fc 2021-12-01)

**Comment by appflowy** [comment]
@xueqiliu @paranoea I have updated the step4 of README. Would you mind checking it again?

https://github.com/AppFlowy-IO/appflowy/blob/main/README.md

**Comment by paranoea** [comment]
@appflowy
Sure. I just got the following log.

Launching lib/main.dart on macOS in debug mode...
lib/main.dart:1
--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:macOS, arch:x86_64, id:0DE14BCB-8D3A-5995-8902-9AB1F39F2711 }
{ platform:macOS, name:Any Mac }
Undefined symbols for architecture x86_64:
  "_link_me_please", referenced from:
      static flowy_sdk.FlowySdkPlugin.dummyMethodToEnforceBundling() -> () in flowy_sdk(FlowySdkPlugin.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
note: Using new build system
note: Planning
note: Build preparation complete
note: Building targets in parallel
** BUILD FAILED **

Exception: Build process failed
Exited (sigterm)

And I attach an image here.
![1638534838230](https://user-images.githubusercontent.com/3395033/144603638-97b2bbf5-3bd5-455e-b207-95aaaaf6b07d.jpg)

**Comment by appflowy** [comment]
Hi, @paranoea would you mind switching to the terminal tag and showing me the log? It seems like the FlowySDK didn't build success.
![image](https://user-images.githubusercontent.com/86001920/144771182-599a8e74-9f38-4747-8621-574afe3f4fe6.png)

**Comment by tortorse** [comment]
The same problem.
I follow the tip and build sdk manually

```shell
sh ../scripts/build_sdk.sh
```

![截屏2021-12-06 09 47 46](https://user-images.githubusercontent.com/10540493/144774571-5d851ae5-d1fe-4084-88fa-a10e3a81bda3.png)

So, I checked the Makefile, I can not find any make about 'flowy-sdk-dev', but I found 'flowy_dev'

**Comment by appflowy** [comment]
@tortorse Are you under the frontend directory? The task is located in appflowy/frontend/scripts/makefile/desktop.toml
![image](https://user-images.githubusercontent.com/86001920/144775411-413c6a7d-b72d-40d6-9a90-b233f212b2f5.png)

**Comment by paranoea** [comment]
@appflowy The flowy sdk was built successfully. But I still got a build break as the same result.

➜  frontend git:(main) ✗ . scripts/build_sdk.sh 
Start building rust sdk
Default host: x86_64-apple-darwin
rustup home:  /Users/paranoia/.rustup

installed toolchains
--------------------

stable-x86_64-apple-darwin
nightly-2021-04-24-x86_64-apple-darwin
nightly-x86_64-apple-darwin (default)

installed targets for active toolchain
--------------------------------------

aarch64-apple-darwin
aarch64-apple-ios
x86_64-apple-darwin
x86_64-apple-ios

active toolchain
----------------

nightly-x86_64-apple-darwin (default)
rustc 1.59.0-nightly (48a5999fc 2021-12-01)

[cargo-make] INFO - cargo make 0.35.6
[cargo-make] INFO - Project: dart-ffi
[cargo-make] INFO - Build File: Makefile.toml
[cargo-make] INFO - Task: flowy-sdk-dev
[cargo-make] INFO - Profile: development-mac
[cargo-make] INFO - Running Task: legacy-migration
[cargo-make] INFO - Running Task: echo_env
-------- Env Parameters --------
CRATE_TYPE: staticlib
BUILD_FLAG: debug
TARGET_OS: macos
RUST_COMPILE_TARGET: x86_64-apple-darwin
FEATURES: flutter
PRODUCT_EXT: app
APP_ENVIRONMENT: local

-------- Flutter --------
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel dev, 2.6.0-11.0.pre, on Mac OS X 10.15.7 19H1519 darwin-x64, locale zh-Hans-CN)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 12.1)
[✓] Chrome - develop for the web
[✓] Android Studio
[!] Android Studio
    ✗ Unable to find bundled Java version.
[✓] Android Studio (version 2020.3)
[✓] VS Code (version 1.62.3)
[✓] Connected device (2 available)

! Doctor found issues in 1 category.
-------- Rust --------
Default host: x86_64-apple-darwin
rustup home:  /Users/paranoia/.rustup

installed toolchains
--------------------

stable-x86_64-apple-darwin
nightly-2021-04-24-x86_64-apple-darwin
nightly-x86_64-apple-darwin (default)

installed targets for active toolchain
--------------------------------------

aarch64-apple-darwin
aarch64-apple-ios
x86_64-apple-darwin
x86_64-apple-ios

active toolchain
----------------

nightly-x86_64-apple-darwin (environment override by RUSTUP_TOOLCHAIN)
rustc 1.59.0-nightly (48a5999fc 2021-12-01)

[cargo-make] INFO - Running Task: setup-crate-type
[cargo-make] INFO - Running Task: sdk-build
cargo build --package=dart-ffi --target x86_64-apple-darwin --features flutter
    Updating crates.io index
  Downloaded async-stream v0.3.2
  Downloaded strum_macros v0.21.1
  Downloaded termcolor v1.1.2
  Downloaded strum v0.21.0
  Downloaded time v0.1.44
  Downloaded signal-hook-registry v1.4.0
  Downloaded thiserror-impl v1.0.30
  Downloaded socket2 v0.4.2
  Downloaded strsim v0.10.0
  Downloaded smallvec v1.7.0
  Downloaded static_assertions v1.1.0
  Downloaded serde-aux v1.1.0
  Downloaded thiserror v1.0.30
  Downloaded tower-service v0.3.1
  Downloaded tokio-util v0.6.9
  Downloaded backtrace v0.3.63
  Downloaded sha-1 v0.9.8
  Downloaded sharded-slab v0.1.4
  Downloaded crossbeam v0.8.1
  Downloaded crossbeam-epoch v0.9.5
  Downloaded crossbeam-queue v0.3.2
  Downloaded dashmap v4.0.2
  Downloaded derivative v2.2.0
  Downloaded derive_more v0.99.17
  Downloaded tracing-attributes v0.1.18
  Downloaded byteorder v1.4.3
  Downloaded httpdate v1.0.2
  Downloaded hyper-tls v0.5.0
  Downloaded tokio-tungstenite v0.15.0
  Downloaded crossbeam-channel v0.5.1
  Downloaded chrono v0.4.19
  Downloaded hyper v0.14.15
  Downloaded bincode v1.3.3
  Downloaded tracing-appender v0.1.2
  Downloaded tokio-native-tls v0.3.0
  Downloaded darling_macro v0.13.0
  Downloaded tracing-bunyan-formatter v0.2.6
  Downloaded memoffset v0.6.5
  Downloaded mime v0.3.16
  Downloaded instant v0.1.12
  Downloaded md5 v0.7.0
  Downloaded humantime v2.1.0
  Downloaded bit-set v0.5.2
  Downloaded migrations_macros v1.4.2
  Downloaded lock_api v0.4.5
  Downloaded gethostname v0.2.1
  Downloaded serde_with_macros v1.5.1
  Downloaded syn v1.0.82
  Downloaded tracing-subscriber v0.2.25
  Downloaded validator_types v0.12.0
  Downloaded tracing-futures v0.2.5
  Downloaded pin-project v1.0.8
  Downloaded futures-sink v0.3.18
  Downloaded try-lock v0.2.3
  Downloaded want v0.3.0
  Downloaded scopeguard v1.1.0
  Downloaded pkg-config v0.3.22
  Downloaded pin-project-lite v0.2.7
  Downloaded regex-automata v0.1.10
  Downloaded tracing v0.1.29
  Downloaded h2 v0.3.7
  Downloaded mio v0.7.14
  Downloaded validator v0.12.0
  Downloaded tracing-serde v0.1.2
  Downloaded protobuf v2.25.2
  Downloaded rustc_version v0.4.0
  Downloaded utf-8 v0.7.6
  Downloaded tracing-core v0.1.21
  Downloaded yaml-rust v0.4.5
  Downloaded serde_urlencoded v0.7.0
  Downloaded reqwest v0.11.7
  Downloaded migrations_internals v1.4.1
  Downloaded tracing-log v0.1.2
  Downloaded uuid v0.8.2
  Downloaded tokio v1.14.0
  Downloaded lexical-core v0.7.6
  Downloaded color-eyre v0.5.11
  Downloaded libc v0.2.109
  Downloaded gimli v0.26.1
  Downloaded diesel_migrations v1.4.0
  Downloaded encoding_rs v0.8.29
  Downloaded libsqlite3-sys v0.22.2
  Downloaded error-chain v0.12.0
  Downloaded rustversion v1.0.5
  Downloaded serde_repr v0.1.7
  Downloaded futures-channel v0.3.18
  Downloaded scheduled-thread-pool v0.2.5
  Downloaded parking_lot v0.11.2
  Downloaded futures-io v0.3.18
  Downloaded r2d2 v0.8.9
  Downloaded linked-hash-map v0.5.4
  Downloaded owo-colors v1.3.0
  Downloaded tungstenite v0.14.0
  Downloaded pin-utils v0.1.0
  Downloaded futures-util v0.3.18
  Downloaded cc v1.0.72
  Downloaded futures v0.3.18
  Downloaded futures-executor v0.3.18
  Downloaded futures-core v0.3.18
  Downloaded futures-task v0.3.18
  Downloaded futures-macro v0.3.18
  Downloaded object v0.27.1
  Downloaded rustc-demangle v0.1.21
  Downloaded proc-macro2 v1.0.33
  Downloaded ryu v1.0.6
  Downloaded pin-project-internal v1.0.8
  Downloaded paste v1.0.6
  Downloaded parking_lot_core v0.8.5
  Downloaded nom v5.1.2
  Downloaded serde_json v1.0.72
  Downloaded matchers v0.0.1
  Downloaded httparse v1.5.1
  Downloaded http-body v0.4.4
  Downloaded ffi-support v0.4.4
  Downloaded fancy-regex v0.5.0
  Downloaded eyre v0.6.5
  Downloaded env_logger v0.8.4
  Downloaded anyhow v1.0.51
  Downloaded ipnet v2.3.1
  Downloaded bit-vec v0.6.3
  Downloaded dyn-clone v1.0.4
  Downloaded diesel_derives v1.4.1
  Downloaded async-stream-impl v0.3.2
  Downloaded darling_core v0.13.0
  Downloaded thread-id v3.3.0
  Downloaded darling v0.13.0
  Downloaded crossbeam-deque v0.8.1
  Downloaded ansi_term v0.12.1
  Downloaded tokio-macros v1.6.0
  Downloaded indenter v0.3.3
  Downloaded ident_case v1.0.1
  Downloaded config v0.10.1
  Downloaded convert_case v0.4.0
  Downloaded serde_with v1.11.0
  Downloaded bytecount v0.6.2
  Downloaded allo-isolate v0.1.12
  Downloaded atomic v0.5.1
  Downloaded arrayvec v0.5.2
  Downloaded addr2line v0.17.0
  Downloaded slab v0.4.5
  Downloaded diesel v1.4.8
  Downloaded 141 crates (11.9 MB) in 13.19s (largest was `libsqlite3-sys` at 2.3 MB)
   Compiling proc-macro2 v1.0.33
   Compiling unicode-xid v0.2.2
   Compiling syn v1.0.82
   Compiling cfg-if v1.0.0
   Compiling libc v0.2.109
   Compiling autocfg v1.0.1
   Compiling memchr v2.4.1
   Compiling serde_derive v1.0.130
   Compiling lazy_static v1.4.0
   Compiling serde v1.0.130
   Compiling log v0.4.14
   Compiling pin-project-lite v0.2.7
   Compiling smallvec v1.7.0
   Compiling scopeguard v1.1.0
   Compiling itoa v0.4.8
   Compiling parking_lot_core v0.8.5
   Compiling ryu v1.0.6
   Compiling once_cell v1.8.0
   Compiling cc v1.0.72
   Compiling futures-core v0.3.18
   Compiling serde_json v1.0.72
   Compiling regex-syntax v0.6.25
   Compiling pkg-config v0.3.22
   Compiling futures-task v0.3.18
   Compiling futures-sink v0.3.18
   Compiling version_check v0.9.3
   Compiling futures-channel v0.3.18
   Compiling futures-util v0.3.18
   Compiling tinyvec_macros v0.1.0
   Compiling matches v0.1.9
   Compiling pin-utils v0.1.0
   Compiling slab v0.4.5
   Compiling percent-encoding v2.1.0
   Compiling futures-io v0.3.18
   Compiling ppv-lite86 v0.2.15
   Compiling core-foundation-sys v0.8.3
   Compiling unicode-bidi v0.3.7
   Compiling protobuf v2.25.2
   Compiling fnv v1.0.7
   Compiling bitflags v1.3.2
   Compiling byteorder v1.4.3
   Compiling typenum v1.14.0
   Compiling httparse v1.5.1
   Compiling adler v1.0.2
   Compiling gimli v0.26.1
   Compiling rustc-demangle v0.1.21
   Compiling convert_case v0.4.0
   Compiling crossbeam-utils v0.8.5
   Compiling unicode-segmentation v1.8.0
   Compiling native-tls v0.2.8
   Compiling ident_case v1.0.1
   Compiling hashbrown v0.11.2
   Compiling remove_dir_all v0.5.3
   Compiling strsim v0.10.0
   Compiling lexical-core v0.7.6
   Compiling try-lock v0.2.3
   Compiling tower-service v0.3.1
   Compiling arrayvec v0.5.2
   Compiling bytecount v0.6.2
   Compiling static_assertions v1.1.0
   Compiling strum v0.21.0
   Compiling base64 v0.13.0
   Compiling rustversion v1.0.5
   Compiling bit-vec v0.6.3
   Compiling encoding_rs v0.8.29
   Compiling httpdate v1.0.2
   Compiling cpufeatures v0.2.1
   Compiling linked-hash-map v0.5.4
   Compiling validator_types v0.12.0
   Compiling md5 v0.7.0
   Compiling anyhow v1.0.51
   Compiling opaque-debug v0.3.0
   Compiling termcolor v1.1.2
   Compiling mime v0.3.16
   Compiling humantime v2.1.0
   Compiling utf-8 v0.7.6
   Compiling paste v1.0.6
   Compiling ipnet v2.3.1
   Compiling crossbeam-epoch v0.9.5
   Compiling dyn-clone v1.0.4
   Compiling ansi_term v0.12.1
   Compiling crossbeam-queue v0.3.2
   Compiling eyre v0.6.5
   Compiling indenter v0.3.3
   Compiling owo-colors v1.3.0
   Compiling instant v0.1.12
   Compiling tracing-core v0.1.21
   Compiling sharded-slab v0.1.4
   Compiling lock_api v0.4.5
   Compiling tokio v1.14.0
   Compiling num-traits v0.2.14
   Compiling num-integer v0.1.44
   Compiling miniz_oxide v0.4.4
   Compiling indexmap v1.7.0
   Compiling atomic v0.5.1
   Compiling memoffset v0.6.5
   Compiling thread_local v1.1.3
   Compiling tinyvec v1.5.1
   Compiling generic-array v0.14.4
   Compiling nom v5.1.2
   Compiling form_urlencoded v1.0.1
   Compiling libsqlite3-sys v0.22.2
   Compiling backtrace v0.3.63
   Compiling heck v0.3.3
   Compiling bit-set v0.5.2
   Compiling regex-automata v0.1.10
   Compiling yaml-rust v0.4.5
   Compiling addr2line v0.17.0
   Compiling want v0.3.0
   Compiling tracing-log v0.1.2
   Compiling ffi-support v0.4.4
   Compiling quote v1.0.10
   Compiling getrandom v0.2.3
   Compiling num_cpus v1.13.0
   Compiling mio v0.7.14
   Compiling signal-hook-registry v1.4.0
   Compiling time v0.1.44
   Compiling socket2 v0.4.2
   Compiling atty v0.2.14
   Compiling thread-id v3.3.0
   Compiling gethostname v0.2.1
   Compiling aho-corasick v0.7.18
   Compiling object v0.27.1
   Compiling unicode-normalization v0.1.19
   Compiling matchers v0.0.1
   Compiling core-foundation v0.9.2
   Compiling security-framework-sys v2.4.2
   Compiling crossbeam-channel v0.5.1
   Compiling rand_core v0.6.3
   Compiling parking_lot v0.11.2
   Compiling dashmap v4.0.2
   Compiling regex v1.5.4
   Compiling idna v0.2.3
   Compiling security-framework v2.4.2
   Compiling digest v0.9.0
   Compiling block-buffer v0.9.0
   Compiling rand_chacha v0.3.1
   Compiling scheduled-thread-pool v0.2.5
   Compiling crossbeam-deque v0.8.1
   Compiling chrono v0.4.19
   Compiling url v2.2.2
   Compiling fancy-regex v0.5.0
   Compiling env_logger v0.8.4
   Compiling error-chain v0.12.0
   Compiling color-eyre v0.5.11
   Compiling flowy-ast v0.1.0 (/Users/paranoia/Projects/projects/appflowy/shared-lib/flowy-ast)
   Compiling darling_core v0.13.0
   Compiling sha-1 v0.9.8
   Compiling rand v0.8.4
   Compiling r2d2 v0.8.9
   Compiling crossbeam v0.8.1
   Compiling tempfile v3.2.0
   Compiling tracing-attributes v0.1.18
   Compiling tokio-macros v1.6.0
   Compiling futures-macro v0.3.18
   Compiling diesel_derives v1.4.1
   Compiling pin-project-internal v1.0.8
   Compiling derive_more v0.99.17
   Compiling thiserror-impl v1.0.30
   Compiling strum_macros v0.21.1
   Compiling derivative v2.2.0
   Compiling serde_repr v0.1.7
   Compiling async-stream-impl v0.3.2
   Compiling flowy-derive v0.1.0 (/Users/paranoia/Projects/projects/appflowy/shared-lib/flowy-derive)
   Compiling pin-project v1.0.8
   Compiling tracing v0.1.29
   Compiling darling_macro v0.13.0
   Compiling async-stream v0.3.2
   Compiling thiserror v1.0.30
   Compiling allo-isolate v0.1.12
   Compiling tracing-futures v0.2.5
   Compiling diesel v1.4.8
   Compiling darling v0.13.0
   Compiling serde_with_macros v1.5.1
   Compiling futures-executor v0.3.18
   Compiling bytes v1.1.0
   Compiling uuid v0.8.2
   Compiling serde_urlencoded v0.7.0
   Compiling serde_with v1.11.0
   Compiling config v0.10.1
   Compiling tracing-serde v0.1.2
   Compiling bincode v1.3.3
   Compiling futures v0.3.18
   Compiling http v0.2.5
   Compiling lib-ot v0.1.0 (/Users/paranoia/Projects/projects/appflowy/shared-lib/lib-ot)
   Compiling validator v0.12.0
   Compiling serde-aux v1.1.0
   Compiling tracing-subscriber v0.2.25
   Compiling migrations_internals v1.4.1
   Compiling flowy-user-infra v0.1.0 (/Users/paranoia/Projects/projects/appflowy/shared-lib/flowy-user-infra)
   Compiling http-body v0.4.4
   Compiling tungstenite v0.14.0
   Compiling migrations_macros v1.4.2
   Compiling tracing-appender v0.1.2
   Compiling tracing-bunyan-formatter v0.2.6
   Compiling lib-log v0.1.0 (/Users/paranoia/Projects/projects/appflowy/frontend/rust-lib/lib-log)
   Compiling diesel_migrations v1.4.0
   Compiling lib-sqlite v0.1.0 (/Users/paranoia/Projects/projects/appflowy/shared-lib/lib-sqlite)
   Compiling tokio-util v0.6.9
   Compiling tokio-native-tls v0.3.0
   Compiling flowy-document-infra v0.1.0 (/Users/paranoia/Projects/projects/appflowy/shared-lib/flowy-document-infra)
   Compiling tokio-tungstenite v0.15.0
   Compiling lib-dispatch v0.1.0 (/Users/paranoia/Projects/projects/appflowy/frontend/rust-lib/lib-dispatch)
   Compiling lib-infra v0.1.0 (/Users/paranoia/Projects/projects/appflowy/frontend/rust-lib/lib-infra)
   Compiling flowy-database v0.1.0 (/Users/paranoia/Projects/projects/appflowy/frontend/rust-lib/flowy-database)
   Compiling h2 v0.3.7
   Compiling dart-notify v0.1.0 (/Users/paranoia/Projects/projects/appflowy/frontend/rust-lib/dart-notify)
   Compiling flowy-workspace-infra v0.1.0 (/Users/paranoia/Projects/projects/appflowy/shared-lib/flowy-workspace-infra)
   Compiling hyper v0.14.15
   Compiling hyper-tls v0.5.0
   Compiling reqwest v0.11.7
   Compiling backend-service v0.1.0 (/Users/paranoia/Projects/projects/appflowy/shared-lib/backend-service)
   Compiling lib-ws v0.1.0 (/Users/paranoia/Projects/projects/appflowy/shared-lib/lib-ws)
   Compiling flowy-document v0.1.0 (/Users/paranoia/Projects/projects/appflowy/frontend/rust-lib/flowy-document)
   Compiling flowy-user v0.1.0 (/Users/paranoia/Projects/projects/appflowy/frontend/rust-lib/flowy-user)
   Compiling flowy-workspace v0.1.0 (/Users/paranoia/Projects/projects/appflowy/frontend/rust-lib/flowy-workspace)
   Compiling flowy-sdk v0.1.0 (/Users/paranoia/Projects/projects/appflowy/frontend/rust-lib/flowy-sdk)
   Compiling dart-ffi v0.1.0 (/Users/paranoia/Projects/projects/appflowy/frontend/rust-lib/dart-ffi)
    Finished dev [unoptimized + debuginfo] target(s) in 3m 11s
[cargo-make] INFO - Running Task: post-desktop
🚀 🚀 🚀  Flowy-SDK(macOS) build success 
[cargo-make] INFO - Running Task: restore-crate-type
[cargo-make] INFO - Build Done in 203.91 seconds.

**Comment by tortorse** [comment]
> @tortorse Are you under the frontend directory? The task is located in appflowy/frontend/scripts/makefile/desktop.toml ![image](https://user-images.githubusercontent.com/86001920/144775411-413c6a7d-b72d-40d6-9a90-b233f212b2f5.png)
> 
> ![image](https://user-images.githubusercontent.com/86001920/144775664-a89ccb3d-ea7d-435d-a48a-a5c5fc359044.png) You could try with this command: `cargo make --profile development-mac flowy-sdk-dev`
> 
> FYI, this command only builds the FlowySDK, not the frontend application.

Thank you. But I got error below:

![image](https://user-images.githubusercontent.com/10540493/144778366-4981475c-381f-4cb5-bac5-b0ba78666a56.png)

**Comment by paranoea** [comment]
@appflowy 
After my successful build on flowy sdk. I can run appflowy now!
I just try run "flutter run" from console.
But the buid break is still there if I press F5 in VS Code. There must be some error in building environment.

**Comment by appflowy** [comment]
@tortorse Maybe you should pull the latest code. It should work

**Comment by appflowy** [comment]
@paranoea I have found some similar issues like yours, but none of them has a solution to fix this. I also notice that you're using Xcode 13.1, I'll update my Xcode to 13.1 and try to reproduce your issues.

https://developer.apple.com/forums/thread/658009
https://askandroidquestions.com/2021/10/07/flutter-xcodebuild-warning-using-the-first-of-multiple-matching-destinations/

Thanks for trying this!

**Comment by paranoea** [comment]
@appflowy Actually, my xcode is 12.1, since I am still working on my Hackintosh Catalina.

**Comment by tortorse** [comment]
@appflowy Thanks, It works!

**Comment by nooop3** [comment]
I can reproduce this error when I build the APP from the terminal. But it works well in the VSCode.
I need to rebuild the MacOS lib to start the app from the terminal, here are the fixing steps:

```bash
FLOWY_DEV_ENV=macOS sh ./scripts/build_sdk.sh
cd app_flowy/macos
pod install
cd ..
flutter run -d macos
```

**Comment by annieappflowy** [comment]
> I can reproduce this error when I build the APP from the terminal. But it works well in the VSCode. I need to rebuild the MacOS lib to start the app from the terminal, here are the fixing steps:
> 
> ```shell
> FLOWY_DEV_ENV=macOS sh ./scripts/build_sdk.sh
> cd app_flowy/macos
> pod install
> cd ..
> flutter run -d macos
> ```

@appflowy

**Comment by saru2020** [comment]
trying from macOS BigSur, not able to build the source on any platform macOS or Chrome, is there a documentation on how to install the app?

**Comment by annieappflowy** [comment]
Hi @saru2020 , https://appflowy.gitbook.io/docs/essential-documentation/install-appflowy/installation-methods/from-source
Let me know if you need more help

**Comment by saru2020** [comment]
thanks for the docs link, @annieappflowy 
I followed the steps and am facing build errors:
```
      ^^^^^^^^^^^^^^^^^^^
: Context: 'TextInputClient.didChangeInputControl' is defined here.
../…/services/text_input.dart:1163
  void didChangeInputControl(TextInputControl? oldControl, TextInputControl? newControl) {}
       ^^^^^^^^^^^^^^^^^^^^^
: Context: 'TextInputClient.performSelector' is defined here.
../…/services/text_input.dart:1184
  void performSelector(String selectorName) {}
       ^^^^^^^^^^^^^^^
Target kernel_snapshot failed: Exception

Command PhaseScriptExecution failed with a nonzero exit code

** BUILD FAILED **

Exception: Build process failed
Exited
```

**Comment by LucasXu0** [comment]
Hey, @saru2020. Can you check if your flutter contains this function or not? `didChangeInputControl `. Because this API should not exist in flutter 3.3.10.

<img width="1624" alt="Screenshot 2023-02-22 at 09 38 18" src="https://user-images.githubusercontent.com/11863087/220498571-1ff6ce7c-7bc7-42e1-9595-883e0fd0e39f.png">

**Comment by saru2020** [comment]
Sorry, that was with the latest Flutter version

but with Flutter 3.3.10 also I'm facing issues:

```
Try correcting the name to the name of an existing getter, or defining a getter or field named 'predicate'.
      predicate: predicate ?? state.predicate,
                                    ^^^^^^^^^
lib/plugins/grid/application/filter/select_option_filter_list_bloc.dart:84:53: Error: The getter 'selectedOptionIds' isn't defined for the class 'SelectOptionFilterListState'.
 - 'SelectOptionFilterListState' is from 'package:app_flowy/plugins/grid/application/filter/select_option_filter_list_bloc.dart' ('lib/plugins/grid/application/filter/select_option_filter_list_bloc.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'selectedOptionIds'.
      selectedOptionIds: selectedOptionIds ?? state.selectedOptionIds,
                                                    ^^^^^^^^^^^^^^^^^
lib/plugins/grid/application/filter/select_option_filter_list_bloc.dart:92:10: Error: 'SelectOptionPB' isn't a type.
    List<SelectOptionPB> options = List.from(state.options);
         ^^^^^^^^^^^^^^
lib/plugins/grid/application/filter/select_option_filter_list_bloc.dart:92:52: Error: The getter 'options' isn't defined for the class 'SelectOptionFilterListState'.
 - 'SelectOptionFilterListState' is from 'package:app_flowy/plugins/grid/application/filter/select_option_filter_list_bloc.dart' ('lib/plugins/grid/application/filter/select_option_filter_list_bloc.dart').
Try correcting the name to the name of an existing getter, or defining a getter or field named 'options'.
    List<SelectOptionPB> options = List.from(state.options);
                                                   ^^^^^^^
lib/plugins/grid/application/filter/select_option_filter_list_bloc.dart:93:46: Error: The getter 'name' isn't defined for the class 'Object?'.
 - 'Object' is from 'dart:core'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'name'.
    options.retainWhere((element) => element.name.contains(predicate));
                                             ^^^^
lib/plugins/grid/application/filter/select_option_filter_list_bloc.dart:96:76: Error: The getter 'id' isn't defined for the class 'Object?'.
 - 'Object' is from 'dart:core'.
Try correcting the name to the name of an existing getter, or defining a getter or field named 'id'.
      return VisibleSelectOption(option, selectedOptionIds.contains(option.id));
                                                                           ^^
lib/plugins/grid/application/filter/select_option_filter_list_bloc.dart:97:8: Error: A value of type 'List<dynamic>' can't be returned from a function with return type 'List<VisibleSelectOption>'.
 - 'List' is from 'dart:core'.
 - 'VisibleSelectOption' is from 'package:app_flowy/plugins/grid/application/filter/select_option_filter_list_bloc.dart' ('lib/plugins/grid/application/filter/select_option_filter_list_bloc.dart').
    }).toList();
       ^
lib/plugins/grid/application/filter/select_option_filter_list_bloc.dart:115:7: Error: 'SelectOptionPB' isn't a type.
      SelectOptionPB option) = _SelectOption;
      ^^^^^^^^^^^^^^
lib/plugins/grid/application/filter/select_option_filter_list_bloc.dart:117:7: Error: 'SelectOptionPB' isn't a type.
      SelectOptionPB option) = _UnSelectOption;
      ^^^^^^^^^^^^^^
lib/plugins/grid/application/filter/select_option_filter_list_bloc.dart:119:12: Error: 'SelectOptionPB' isn't a type.
      List<SelectOptionPB> options) = _DidReceiveOptions;
           ^^^^^^^^^^^^^^
lib/plugins/grid/application/filter/select_option_filter_list_bloc.dart:127:19: Error: 'SelectOptionPB' isn't a type.
    required List<SelectOptionPB> options,
                  ^^^^^^^^^^^^^^
lib/plugins/grid/application/filter/select_option_filter_list_bloc.dart:144:9: Error: 'SelectOptionPB' isn't a type.
  final SelectOptionPB optionPB;
        ^^^^^^^^^^^^^^

Command PhaseScriptExecution failed with a nonzero exit code
** BUILD FAILED **

Building macOS application...                                           
Exception: Build process failed
```
```flutter --version``` output:
```
Flutter 3.3.10 • channel unknown • unknown source
Framework • revision 135454af32 (10 weeks ago) • 2022-12-15 07:36:55 -0800
Engine • revision 3316dd8728
Tools • Dart 2.18.6 • DevTools 2.15.0
```

**Comment by Xazin** [comment]
As this has become out-of-date, I'm going to close this.

If you're still experiencing issues, please create a new issue with information, and we'll do our best to help you resolve it.
